Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/network/NetworkStateProvider.kt

```diff
@@ -90,34 +92,46 @@ class NetworkStateProvider @Inject constructor(
         )
 
     private val currentState: State
-        get() = manager.activeNetwork.let { network ->
+        @SuppressLint("NewApi")
+        get() = if (BuildVersionWrap.hasAPILevel(Build.VERSION_CODES.M)) {
+            manager.activeNetwork.let { network ->
+                State(
+                    activeNetwork = network,
+                    capabilities = network?.let {
+                        try {
+                            manager.getNetworkCapabilities(it)
+                        } catch (e: SecurityException) {
+                            Timber.tag(TAG).e(e, "Failed to determine network capabilities.")
+                            null
+                        }
+                    },
+                    linkProperties = network?.let {
+                        try {
+                            manager.getLinkProperties(it)
+                        } catch (e: Exception) {
+                            Timber.tag(TAG).e(e, "Failed to determine link properties.")
+                            null
+                        }
+                    },
+                    assumeMeteredConnection = testSettings.fakeMeteredConnection.value
+                )
+            }
+        } else {
+            // Most state information is not available
```

Comment:
Maybe we can split it into small functions and adjusted it a bit ?

```kotlin
 private val currentState: State
        get() = when {
            Build.VERSION.SDK_INT >= Build.VERSION_CODES.M -> api23NetworkState()
            else -> {
                // Most state information is not available
                State(
                    activeNetwork = null,
                    capabilities = null,
                    linkProperties = null,
                    assumeMeteredConnection = testSettings.fakeMeteredConnection.value ||
                        ConnectivityManagerCompat.isActiveNetworkMetered(manager)
                )
            }
        }

    @RequiresApi(Build.VERSION_CODES.M)
    private fun api23NetworkState() = manager.activeNetwork.let { network ->
        State(
            activeNetwork = network,
            capabilities = network?.let {
                try {
                    manager.getNetworkCapabilities(it)
                } catch (e: SecurityException) {
                    Timber.tag(TAG).e(e, "Failed to determine network capabilities.")
                    null
                }
            },
            linkProperties = network?.let {
                try {
                    manager.getLinkProperties(it)
                } catch (e: Exception) {
                    Timber.tag(TAG).e(e, "Failed to determine link properties.")
                    null
                }
            },
            assumeMeteredConnection = testSettings.fakeMeteredConnection.value
        )
    }
```

Reply:
A problem here is that `BuildVersionWrap.SDK_INT` can be mocked in the tests while this is not possible (or I don't know how) when comparing the value directly. Therefore I changed this again to use `BuildVersionWrap` and `@SuppressLint`.